pkg/controller/middleware/auth.go

```diff
@@ -165,22 +127,32 @@ func (h *RequireAuthHandler) Handle(next http.Handler) http.Handler {
 		ctx := r.Context()
 
 		if err := func() error {
-			// Get the cookie
-			cookie, err := r.Cookie("session")
+			// Get the session
+			session, err := h.sessions.Get(r, "session")
 			if err != nil {
-				return fmt.Errorf("failed to get cookie: %w", err)
+				return fmt.Errorf("failed to get session: %w", err)
+			}
+
+			// Verify firebase cookie value is present
+			if session.Values == nil || session.Values["firebaseCookie"] == nil {
```

Comment:
This is a somewhat annoying side effect of gorilla sessions. Values can be nil and Values is a `map[interface{}][interface{}]`, so the key can also be nil.